pkg/ingester/streams_map.go

```diff
@@ -45,6 +45,7 @@ func (m *streamsMap) StoreByFP(fp model.Fingerprint, s *stream) {
 
 // Delete must be called inside WithLock
 func (m *streamsMap) Delete(s *stream) bool {
+	s.Stop()
```

Comment:
We only want to do this if `loaded == true`

Reply:
hmm are you sure? I considered it but since we'll always have the rateLimiter operating for a stream (even if it was not loaded), I felt this was necessary for all cases. I'll investigate what happens in such cases.

Reply:
That's a fair point. `LoadAndDelete` returning false means the stream wasn't in the map. Is it even possible to delete a stream that's not in the map?